### ISSUE TITLE
librouting.so: undefined symbol: HeapTupleHeaderGetDatum

### ISSUE THREAD

**Issue by marcinzp** [pgRouting/pgrouting#271]
Hello, I'm trying to build pgrouting for Debian 7.5 and PostgreSQL 9.3
Everything builds and installs OK, but:

```
# create extension pgrouting;
ERROR:  could not load library "/usr/lib/postgresql/9.3/lib/librouting.so": /usr/lib/postgresql/9.3/lib/librouting.so: undefined symbol: HeapTupleHeaderGetDatum
```
 ldd /usr/lib/postgresql/9.3/lib/librouting.so
	linux-vdso.so.1 =>  (0x00007fff8edff000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f8a6096b000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f8a606e9000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f8a604d2000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f8a60147000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f8a60ebe000)

I tried different source debs for this and future Debian releases, I've build pgrouting manually from git sources also, to no avail. How can I fix it?

regards
Marcin

**Comment by robe2** [comment]
Funny how I came over this issue after so much time has passed trying to find issue to my same problem.  I just ran into the same issue with building for Windows.  It turns out that between PostgreSQL 9.2.8 and 9.2.9 this binding was introduced.  So when I tried to build a custom build for a windows user, using the latest PostgreSQL 9.2.9, it didn't work for him or me.  After some sleuthing,  we realized the issue was that PostGIS (and probably similarly pgRouting) built against 9.2.9 doesn't work in a 9.2.8 or lower instance.  This  in theory should never happen since micros are supposed to be stable and backward compatible with micros of same minor.

Anyway our particular issue is noted here:  http://trac.osgeo.org/postgis/ticket/2995#comment:13
and discussed in PostGIS IRC channel:
http://irclogs.geoapt.com/postgis/%23postgis.2015-01-30.log

SO if you are running PostgreSQL 9.3, it is possible a similar thing happened for 9.3 since dbaston thought he saw this new HeapTupleHeaderGetDatum in May 2014 (which means it might not be in 9.3.0 or a beta)

So if your PostgreSQL 9.3 instance was older, it might not work with a newer postgresql 9.3-dev package  That would mean you need to upgrade your PostgreSQL running version as well.

**Comment by woodbri** [comment]
I think we will need specific instructions to reproduce this issues on ubuntu or debian. Since this seems to be postgresql version changes that violate their own compatibility rules, I'm not sure if it will be possible to use to fix it in pgRouting. It is more likely that the system package builders will need to detect this at the package level.

@robe2 do you have any thoughts on this since you doing the Windows packaging?

**Comment by cvvergara** [comment]
Since 2.1 we only use:
HeapTupleGetDatum
instead of:
HeapTupleHeaderGetDatum
So it was fixed on 2.1
I'll mark it as fixed in develop (milestone 2.1 is closed)
